src/main/java/net/minecraftforge/event/datafix/ConfigureDataFixSchemaEvent.java

```diff
@@ -0,0 +1,208 @@
+    private final Map<String, Supplier<TypeTemplate>> blockEntityTypesView = Collections.unmodifiableMap(blockEntityTypes);
+
+    public ConfigureDataFixSchemaEvent(final Schema schema)
+    {
+        this.schema = schema;
+    }
+
+    /**
+     * The schema which you can configure.
+     *
+     * @return The schema
+     */
+    public Schema getSchema()
+    {
+        return schema;
+    }
+
+    /**
+     * The version of the schema that is about to be configured.
+     *
+     * @return The version.
+     */
+    public int getVersion()
+    {
+        return getSchema().getVersionKey();
+    }
+
+    /**
+     * Registers a new simple (catch-all-nbt) block entity to the current schema and all its descendent versions.
+     *
+     * @param registryName The registry name of the block entity.
+     */
+    public void registerSimpleBlockEntity(final ResourceLocation registryName)
+    {
+        registerBlockEntity(registryName, DSL::remainder);
+    }
+
+    /**
+     * Registers a new simple (catch-all-nbt) block entity with an item inventory array on the nbt field with the given name to the current schema and all its descendent versions.
+     *
+     * @param registryName The registry name of the block entity.
+     * @param nbtKey The name of the nbt field to where the item array is located.
+     */
+    public void registerSimpleItemInventory(final ResourceLocation registryName, final String nbtKey)
+    {
+        registerBlockEntity(registryName, () -> DSL.optionalFields(nbtKey, DSL.list(References.ITEM_STACK.in(getSchema()))));
+    }
+
+    /**
+     * Allows for the custom definition of a new block entity type and registration of it to the current schema and all its descendent versions.
+     * The function is given the registry name of the block entity type template to produce.
+     *
+     * @param registryName The registry name of the block entity type.
+     * @param template The function to produce the block entity type template.
+     */
+    public void registerBlockEntity(final ResourceLocation registryName, final Function<String, TypeTemplate> template)
+    {
+        registerBlockEntity(registryName, () -> template.apply(registryName.toString()));
+    }
+
+    /**
+     * Allows for the custom definition of a new block entity type and registration of it to the current schema and all its descendent versions.
+     * The supplier is invoked to produce the block entity type template.
+     *
+     * @param registryName The registry name of the block entity type.
+     * @param template The supplier of the type template.
+     */
+    public void registerBlockEntity(final ResourceLocation registryName, final Supplier<TypeTemplate> template)
+    {
+        blockEntityTypes.put(registryName.toString(), template);
+    }
+
+    /**
+     * Removes an already registered block entity from the current schema and all its descendent versions.
```

Comment:
Should this be block entity type instead of just block entity?

Reply:
No this does not refer to the type, but to the NBT of the actual block entity.